modules/consignment-api/database.tf

```diff
@@ -21,6 +21,22 @@ resource "random_string" "snapshot_prefix" {
   special = false
 }
 
+resource "aws_kms_key" "encryption" {
+  description         = "KMS key for encryption within the environment"
+  enable_key_rotation = true
```

Comment:
As discussed in person, I've looked into what key rotation means for RDS instances.

When the KMS key is rotated, [the DB is _not_ rencrypted](https://forums.aws.amazon.com/message.jspa?messageID=770958#770910). The data continues to be encrypted with the original value of the CMK. KMS stores all previous values of the CMK so you can keep decrypting old data.

I'm a bit confused about whether RDS uses the CMK itself, or a data key encrypted by the CMK. Most of the docs I've seen imply the former ([example](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Overview.Encryption.html)) but [this diagram](https://aws.amazon.com/rds/features/security/#Encryption_of_Data_at_Rest) implies the latter. But I think the upshot is the same: the RDS DB is never re-encrypted, unless you do it manually by creating and restoring a snapshot using a new key.

Reply:
Thanks for researching this and clarifying!